### DISCUSSION TITLE
Custom Components

### DISCUSSION THREAD

**Discussion by Fidel-C** [piccolo-orm/piccolo_admin#306] [Ideas]
I see that on the long run I may need to add custom components like bar/pie charts , graphs, etc to my project, as these are the demands of many clients today.

Is there any way to modify Piccolo admin templates to include this type of components,  using Chartsjs?

**Comment by dantownsend** [comment]
At some point we might add builtin support for charts etc.

But it's unlikely we'll be able to solve every use case, so we need something more flexible.

So some things we could do is:

1. Allow users to specify links in the sidebar which link out to other URLs which contain charts.
2. Allow iframes to be embedded.
3. Allow custom endpoints / templates to be attached.

I'm quite keen on number 1 as an initial basic solution, as it's achievable quite quickly, and I think it's useful for a few things, and then adding the other stuff later on. What do you think?

**Reply by Fidel-C** [reply]
If  allowing users to specify links  in the sidebar is  quickly achievable then it's very good for a start.

**Reply by dantownsend** [reply]
OK, cool - will create a ticket for that.

**Reply by dantownsend** [reply]
https://github.com/piccolo-orm/piccolo_admin/issues/307
